lib/sles4sap_publiccloud.pm

```diff
@@ -741,4 +794,145 @@ sub create_hana_vars_section {
     return (\%hana_vars);
 }
 
+=head2 setup_azure_native_fencing_permissions
+
+    setup_azure_native_fencing_permissions(vmname=>$vm_name,
+        subscription_id=>$subscription_id,
+        resource_group=>$resource_group,
+        role=>$role);
+
+    Sets up managed identity (MSI) by enabling system assigned identity and
+        role - role to be assigned
+        resource_group - resource group resource belongs to
+        subscription_id - valid azure subscription
+=cut
+
+sub setup_azure_native_fencing_permissions {
+    my (%args) = @_;
+    my $vm_name = $args{vm_name};
+    my $subscription_id = $args{subscription_id};
+    my $resource_group = $args{resource_group};
+    my $role = 'Virtual Machine Contributor';
+
+    foreach ('vm_name', 'subscription_id', 'resource_group') {
+        croak "Missing argument: '$_'" unless defined($args{$_});
+    }
+
+    my $vm_id = azure_enable_system_assigned_identity($vm_name, $resource_group);
+    azure_assign_role(assignee => $vm_id, role => $role, subscription_id => $subscription_id, resource_group => $resource_group);
+    return 1;
+}
+
+=head2 azure_enable_system_assigned_identity
+
+    azure_enable_system_assigned_identity($vm_name, $resource_group);
+
+    Enables 'System assigned identity' for specified VM.
+    Returns 'systemAssignedIdentity' ID.
+
+=cut
+
+sub azure_enable_system_assigned_identity {
```

Comment:
This one seems to me almost pure `az` composition and execution. If it is so I suggest to:
- rename it to  `qesap_az_enable_system_assigned_identity`
- move it to `lib/qesapdeployment.pm`
- test it in `t/15_qesap_azure.t`

Reply:
Moved, thanks.